### PR TITLE
jit(arm64): support for conversion instructions

### PR DESCRIPTION
**Pull request by mathetake**
part of #187

### CODE REVIEW

**Comment by mathetake** [comment]
ok the left is the notorious ITruncFromF and FConvertFromI..

**Comment by mathetake** [comment]
ok done!!

File: internal/wasm/jit/jit_arm64.go [hunk 1]
```diff
@@ -2049,43 +2050,143 @@ func (c *arm64Compiler) compileCopysign(o *wazeroir.OperationCopysign) error {
...
 
+// compileF64ReinterpretFromI64 implements compiler.compileF64ReinterpretFromI64 for the arm64 architecture.
 func (c *arm64Compiler) compileF64ReinterpretFromI64() error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	return c.compileSimpleConversion(arm64.AFMOVD, generalPurposeRegisterTypeFloat)
+}
+
+func (c *arm64Compiler) compileSimpleConversion(inst obj.As, destinationRegType generalPurposeRegisterType) error {
+	if peek := c.locationStack.peek(); peek.onStack() {
+		// If the value is on the stack, this is no-op as there is nothing to do for converting type.
```
[the remark this answers is not in the document]
Comment (mathetake) [reply internal/wasm/jit/jit_arm64.go:2153]: oh right! good catch

**Review by nullpo-head — APPROVED** [review]
LGTM now, thanks!

File: internal/wasm/jit/jit_arm64.go [hunk 2]
```diff
@@ -2049,43 +2050,157 @@ func (c *arm64Compiler) compileCopysign(o *wazeroir.OperationCopysign) error {
 }
 
 func (c *arm64Compiler) compileI32WrapFromI64() error {
```
Comment (codefromthecrypt) [inline internal/wasm/jit/jit_arm64.go:2052]:
```suggestion
// compileI32WrapFromI64 implements compiler.compileI32WrapFromI64 for the arm64 architecture.
func (c *arm64Compiler) compileI32WrapFromI64() error {
```

File: internal/wasm/jit/jit_arm64.go [hunk 3]
```diff
@@ -2049,43 +2050,157 @@ func (c *arm64Compiler) compileCopysign(o *wazeroir.OperationCopysign) error {
...
 func (c *arm64Compiler) compileExtend(o *wazeroir.OperationExtend) error {
-	return fmt.Errorf("TODO: unsupported on arm64")
+	if o.Signed {
+		return c.compileSimpleUniop(arm64.ASXTW)
+	} else {
+		return c.compileSimpleUniop(arm64.AUXTW)
+	}
+}
+
+func (c *arm64Compiler) compileSimpleUniop(inst obj.As) error {
```
Comment (codefromthecrypt) [inline internal/wasm/jit/jit_arm64.go:2195]: compileSimpleUniop is uniop a thing?

File: internal/wasm/jit/jit_arm64_test.go [hunk 4]
```diff
@@ -3370,3 +3369,453 @@ func TestArm64Compiler_compile_Min_Max_Copysign(t *testing.T) {
...
+func TestArm64Compiler_compileReinterpret(t *testing.T) {
+	for _, kind := range []wazeroir.OperationKind{
+		wazeroir.OperationKindF32ReinterpretFromI32,
+		wazeroir.OperationKindF64ReinterpretFromI64,
+		wazeroir.OperationKindI32ReinterpretFromF32,
+		wazeroir.OperationKindI64ReinterpretFromF64,
+	} {
+		kind := kind
+		t.Run(kind.String(), func(t *testing.T) {
+			for _, originOnStack := range []bool{false, true} {
```
Comment (codefromthecrypt) [inline internal/wasm/jit/jit_arm64_test.go:3470]: good dimension

**Review by codefromthecrypt — APPROVED** [review]
almost there

**Pull request merged**